File: src/main/java/fr/lewon/web/bot/entities/input/battle/ActionBattleMob.java
package fr.lewon.web.bot.entities.input.battle;

import fr.lewon.bot.http.body.urlencoded.FUEMember;
import fr.lewon.web.bot.entities.battle.BattleMob;

public class ActionBattleMob extends ActionBattle {

	@FUEMember("who[id_troll]")
	private String whoIdTroll;

	@FUEMember("who[id_world]")
	private String whoIdWorld;

	public ActionBattleMob(BattleMob battleMob) {
		super(battleMob);
		this.whoIdTroll = battleMob.getIdTroll();
		this.whoIdWorld = battleMob.getIdWorld();
	}
}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/battle/ActionBattleMob.kt
package fr.lewon.bot.hh.entities.input.battle

import com.fasterxml.jackson.annotation.JsonProperty
import fr.lewon.bot.hh.entities.battle.BattleMob

class ActionBattleMob(battleMob: BattleMob) : ActionBattle(battleMob) {
    @JsonProperty("who[id_troll]")
    val whoIdTroll: String? = battleMob.idTroll
    @JsonProperty("who[id_world]")
    val whoIdWorld: String? = battleMob.idWorld

}